src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2106,14 +2121,45 @@ int wm_vuldet_report_agent_vulnerabilities(sqlite3 *db, agent_software *agent, w
     return retval;
 }
 
-int wm_vuldet_check_agent_vulnerabilities(agent_software *agents, wm_vuldet_flags *flags, time_t min_full_scan_interval) {
-    agent_software *agents_it;
+
+int wm_vuldet_find_obsolete_vulnerabilities(scan_ctx_t* scan_ctx) {
+    int sock = wm_vuldet_get_wdb_socket();
+    cJSON* j_obsolete_vulns = NULL;
+    cJSON *j_vuln = NULL;
+
+    if (scan_ctx->scan_type == VU_BASELINE_SCAN) {
```

Comment:
Shouldn't this check be done by the caller? I think that this method shouldn't have knowledge related to the scan types.

Reply:
As **VU_BASELINE_SCAN** will occur only once, I think leaving it outside will add complexity when reading `wm_vuldet_check_agent_vulnerabilities`, but it's true that isn't required to be checked here. What do you think?

Reply:
Let's keep it as it is implemented now.